Types/ELF/include/elf.hpp

```diff
@@ -0,0 +1,419 @@
+
+    struct FstEntry64
+    {
+        uint64 pc;
+        uint32 functionOffset;
+    };
+
+    struct Func32
+    {
+        uint32 entry;    // start pc
+        int32 name;      // name (offset to C string)
+        int32 args;      // size of arguments passed to function
+        int32 frame;     // size of function frame, including saved caller PC
+        int32 pcsp;      // pcsp table (offset to pcvalue table)
+        int32 pcfile;    // pcfile table (offset to pcvalue table)
+        int32 pcln;      // pcln table (offset to pcvalue table)
+        int32 nfuncdata; // number of entries in funcdata list
+        int32 npcdata;   // number of entries in pcdata list
+    };
+
+    struct Func64
+    {
+        uint64 entry;    // start pc
+        int32 name;      // name (offset to C string)
+        int32 args;      // size of arguments passed to function
+        int32 frame;     // size of function frame, including saved caller PC
+        int32 pcsp;      // pcsp table (offset to pcvalue table)
+        int32 pcfile;    // pcfile table (offset to pcvalue table)
+        int32 pcln;      // pcln table (offset to pcvalue table)
+        int32 nfuncdata; // number of entries in funcdata list
+        int32 npcdata;   // number of entries in pcdata list
+    };
+} // namespace Go
+
+enum class AddressType : uint8
+{
+    FileOffset = 0,
+    VA         = 1
+};
+
+constexpr auto ELF_INVALID_ADDRESS = 0xFFFFFFFFFFFFFFFF;
+
+namespace Panels
+{
+    enum class IDs : uint8_t
+    {
+        Information    = 0x0,
+        Segments       = 0x1,
+        Sections       = 0x2,
+        GoInformation  = 0x3,
+        StaticSymbols  = 0x4,
+        DynamicSymbols = 0x5,
+    };
+};
+
+class ELFFile : public TypeInterface, public GView::View::BufferViewer::OffsetTranslateInterface
+{
+  public:
+    uint64 panelsMask{ 0 };
+
+    Elf32_Ehdr header32;
+    Elf64_Ehdr header64;
+    bool is64{ false };
+
+    std::vector<Elf32_Phdr> segments32;
+    std::vector<Elf64_Phdr> segments64;
+
+    std::vector<Elf32_Shdr> sections32;
+    std::vector<Elf64_Shdr> sections64;
+
+    std::vector<std::string> sectionNames;
+    std::vector<uint32> sectionsToSegments;
+
+    std::vector<Elf32_Sym> staticSymbols32;
+    std::vector<Elf64_Sym> staticSymbols64;
+    std::vector<std::string> staticSymbolsNames;
+
+    std::vector<Elf32_Sym> dynamicSymbols32;
+    std::vector<Elf64_Sym> dynamicSymbols64;
+    std::vector<std::string> dynamicSymbolsNames;
+
+    // GO
+    uint32 nameSize = 0;
+    uint32 valSize  = 0;
+    uint32 tag      = 0;
+    std::string noteName{};
+    std::string buildId;
+    std::string gnuString;
+    uint32 goPlcntabSectionIndex = -1;
+    Go::GoPclntab112 pclntab112{};
+    std::vector<Go::FstEntry32> entries32;
+    std::vector<Go::FstEntry64> entries64;
+    std::vector<Go::Func32> functions32;
+    std::vector<Go::Func64> functions64;
+    std::vector<std::string> functionsNames;
+
+  public:
+    ELFFile();
+    virtual ~ELFFile()
```

Comment:
sigur e mai bine sa pui asta intr-un cpp

Reply:
ce anume?

Reply:
destructorul sa nu fie in header

Reply:
ok, will change